### PR TITLE
support tailing-sidecar multi-arch image build 

### PR DESCRIPTION
**Pull request by JaredTan95**
it works well in my fork repo actions ci: https://github.com/JaredTan95/tailing-sidecar/actions/runs/3541915453/jobs/5946774315

### CODE REVIEW

**Comment by JaredTan95** [comment]
ops, seems there is not permissions of github packages in `dev builds`: https://github.com/SumoLogic/tailing-sidecar/actions/runs/3542022775/jobs/5947012316#step:7:12

File: .github/workflows/dev_builds.yml [hunk 1]
```diff
@@ -5,10 +5,14 @@ on:
     branches:
       - main
       - 'release-v[0-9]+.[0-9]+'
+  pull_request:
+    branches:
+      - main
+      - 'release-v[0-9]+.[0-9]+'
```
Comment (swiatekm-sumo) [inline .github/workflows/dev_builds.yml:11]: I think this was just for testing, and should be removed?

Comment (swiatekm-sumo) [reply .github/workflows/dev_builds.yml:11]: If you want to build multiplatform images for each PR, you need to add it here: https://github.com/SumoLogic/tailing-sidecar/blob/main/.github/workflows/pull_requests.yml. What you've done here is make us build *and push* the image for each PR, which we definitely don't want.

Comment (JaredTan95) [reply .github/workflows/dev_builds.yml:11]: Ok, I will remove it.

File: .github/workflows/dev_builds.yml [hunk 2]
```diff
@@ -5,10 +5,11 @@ on:
     branches:
       - main
       - 'release-v[0-9]+.[0-9]+'
+
 env:
   SIDECAR_IMAGE: "ghcr.io/sumologic/tailing-sidecar"
   OPERATOR_IMAGE: "ghcr.io/sumologic/tailing-sidecar-operator"
-  LATEST_TAG: "main"
+  TAG: ${{ github.sha }}
```
Comment (swiatekm-sumo) [inline .github/workflows/dev_builds.yml:12]: This doesn't look like it's used anywhere.

File: .github/workflows/dev_builds.yml [hunk 3]
```diff
@@ -20,22 +21,20 @@ jobs:
...
           password: ${{ secrets.CR_PAT }}
-      - name: Push tailing sidecar image
-        run: make push TAG=${{ env.SIDECAR_IMAGE }}:${{ steps.extract_tag.outputs.tag }}
-        working-directory: ./sidecar
-      - name: Set latest image for tailing sidecar
-        run: docker tag ${{ env.SIDECAR_IMAGE }}:${{ steps.extract_tag.outputs.tag }} ${{ env.SIDECAR_IMAGE }}:${{ env.LATEST_TAG }}
-      - name: Push tailing sidecar image with latest tag
-        run: make push TAG=${{ env.SIDECAR_IMAGE }}:${{ env.LATEST_TAG }}
+      - name: Build and push tailing sidecar multiplatform image
+        run: make build-push-multiplatform IMG=${{ env.SIDECAR_IMAGE }}:${{ steps.extract_tag.outputs.tag }}
```
Comment (swiatekm-sumo) [inline .github/workflows/dev_builds.yml:37]: If we're changing the tagging logic, can you also tag this with `latest`?

**Review by swiatekm-sumo — COMMENTED** [review]
Some more comments.

**Comment by JaredTan95** [comment]
@swiatekm-sumo hi, do you have time to help review this?

**Review by swiatekm-sumo — APPROVED** [review]
LGTM. Sorry for not noticing this sooner.

File: .github/workflows/dev_builds.yml [hunk 4]
```diff
@@ -50,24 +53,26 @@ jobs:
...
-        run: make docker-push IMG=${{ env.OPERATOR_IMAGE }}:${{ steps.extract_tag.outputs.tag }}
+      - name: Build and push tailing sidecar operator multiplatform image
+        run: make build-push-multiplatform IMG=${{ env.OPERATOR_IMAGE }}:${{ steps.extract_tag.outputs.tag }}
         working-directory: ./operator
-      - name: Set latest image for tailing sidecar operator
-        run: docker tag ${{ env.OPERATOR_IMAGE }}:${{ steps.extract_tag.outputs.tag }} ${{ env.OPERATOR_IMAGE }}:${{ env.LATEST_TAG }}
-      - name: Push tailing sidecar operator image with latest tag
+      - name: Set latest image for tailing sidecar operator	
+        run: docker tag ${{ env.OPERATOR_IMAGE }}:${{ steps.extract_tag.outputs.tag }} ${{ env.OPERATOR_IMAGE }}:${{ env.LATEST_TAG }}	
+      - name: Push tailing sidecar operator image with latest tag
```
Comment (sumo-drosiek) [inline .github/workflows/dev_builds.yml:74]:
```suggestion
      - name: Set latest image for tailing sidecar operator
        run: docker tag ${{ env.OPERATOR_IMAGE }}:${{ steps.extract_tag.outputs.tag }} ${{ env.OPERATOR_IMAGE }}:${{ env.LATEST_TAG }}	
      - name: Push tailing sidecar operator image with latest tag
```

File: .github/workflows/dev_builds.yml [hunk 5]
```diff
@@ -50,24 +53,26 @@ jobs:
...
           username: USERNAME
           password: ${{ secrets.CR_PAT }}
-      - name: Push tailing sidecar operator image
-        run: make docker-push IMG=${{ env.OPERATOR_IMAGE }}:${{ steps.extract_tag.outputs.tag }}
+      - name: Build and push tailing sidecar operator multiplatform image
+        run: make build-push-multiplatform IMG=${{ env.OPERATOR_IMAGE }}:${{ steps.extract_tag.outputs.tag }}
         working-directory: ./operator
       - name: Set latest image for tailing sidecar operator
-        run: docker tag ${{ env.OPERATOR_IMAGE }}:${{ steps.extract_tag.outputs.tag }} ${{ env.OPERATOR_IMAGE }}:${{ env.LATEST_TAG }}
+        run: docker tag ${{ env.OPERATOR_IMAGE }}:${{ steps.extract_tag.outputs.tag }} ${{ env.OPERATOR_IMAGE }}:${{ env.LATEST_TAG }}
```
Comment (sumo-drosiek) [inline .github/workflows/dev_builds.yml:73]:
```suggestion
        run: docker tag ${{ env.OPERATOR_IMAGE }}:${{ steps.extract_tag.outputs.tag }} ${{ env.OPERATOR_IMAGE }}:${{ env.LATEST_TAG }}
```

**Review by sumo-drosiek — APPROVED** [review]

**Comment by sumo-drosiek** [comment]
@JaredTan95 Sorry for two additional commits, please squash all commits and the change can be merged

**Comment by JaredTan95** [comment]
> @JaredTan95 Sorry for two additional commits, please squash all commits and the change can be merged
> 
> nvm, I made it for you :)

thx

**Pull request merged**